src/app.ts

```diff
@@ -146,6 +146,10 @@ if (cluster.isPrimary) {
 				}`,
 			);
 			_.times(VPN_INSTANCE_COUNT, (i) => {
+				const drainCount = Object.keys(workerStates).length;
```

Comment:
@Page- do you happen to know how `drainCount` is resolved here?

I always get:

```
Jun 27 23:18:01 open-balena-vpn-589468c9ff-zr4kz vpn[530]: info: [master-50933] 0/4 instances draining
Jun 27 23:18:01 open-balena-vpn-589468c9ff-zr4kz vpn[530]: info: [master-50933] 0/4 instances draining
Jun 27 23:18:01 open-balena-vpn-589468c9ff-zr4kz vpn[530]: info: [master-50933] 0/4 instances draining
Jun 27 23:18:01 open-balena-vpn-589468c9ff-zr4kz vpn[530]: info: [master-50933] 0/4 instances draining
```

... which suggests a copy of `workerStates` is taken at init only. If that's the case, do I need to pass a pointer to it somehow to the service.wrap() function?

Reply:
I've changed tactic a bit and trying to route every interrupt in the worker to the drain function, including the `unhandledException`.